DSCResources/MSFT_xDNSServerAddress/MSFT_xDNSServerAddress.psm1

```diff
@@ -87,73 +105,86 @@ function Set-TargetResource
         $($LocalizedData.ApplyingDNSServerAddressesMessage)
         ) -join '')
 
-    #Get the current DNS Server Addresses based on the parameters given.
-    $PSBoundParameters.Remove('Address')
-    $PSBoundParameters.Remove('Validate')
-    $currentAddress = (Get-DnsClientServerAddress @PSBoundParameters `
-        -ErrorAction Stop).ServerAddresses
-
-    #Check if the Server addresses are the same as the desired addresses.
-    [Boolean] $addressDifferent = (@(Compare-Object `
-            -ReferenceObject $currentAddress `
-            -DifferenceObject $Address `
-            -SyncWindow 0).Length -gt 0)
-
-    if ($addressDifferent)
+    # Get the current DNS Server Addresses based on the parameters given.
+    # Interface Alias may return multiple when passed WildCard (*)
+    $interfaceAliases = Get-NetAdapter -Name $InterfaceAlias
+    foreach ($alias in $interfaceAliases.InterfaceAlias)
     {
-        # Set the DNS settings as well
-        $Splat = @{
-            InterfaceAlias = $InterfaceAlias
-            Address = $Address
-            Validate = $Validate
+        $currentAddress = (Get-DnsClientServerAddress `
+                -InterfaceAlias $alias `
+                -AddressFamily $AddressFamily `
+                -ErrorAction Stop).ServerAddresses
+
+        # Check if the Server addresses are the same as the desired addresses.
+        [Boolean] $addressDifferent = (@(Compare-Object `
```

Comment:
Thinking through this again, do we really need to do this? If Test was true then we wouldn't be running set. As we wouldn't be running set do we need this if/else logic? If not could we just set the DNS Client and be done with it?